rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/common/RMRestInterface.java

```diff
@@ -1077,4 +1077,21 @@ void setNodeTokens(@HeaderParam("sessionid") String sessionId, @HeaderParam("nod
     @Produces(MediaType.APPLICATION_JSON)
     Set<String> searchNodes(@HeaderParam("sessionid") String sessionId, @QueryParam("tags") List<String> tags,
             @QueryParam("all") @DefaultValue("true") boolean all) throws NotConnectedException, RestException;
+
+    /**
+     * Check is the user has admin permission for the given nodes or it is a node source provider
```

Comment:
Check if the user...

The request should be renamed checkNodesAdminPermission, the endpoint can remain like this.